ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadTask.java

```diff
@@ -522,6 +525,41 @@ private int executeIncrementalLoad(DriverContext driverContext) {
       // bootstrap of tables if exist.
       if (builder.hasMoreWork() || work.getPathsToCopyIterator().hasNext() || work.hasBootstrapLoadTasks()) {
         DAGTraversal.traverse(childTasks, new AddDependencyToLeaves(TaskFactory.get(work, conf)));
+      } else {
+        // Nothing to be done for repl load now. Add a task to update the last.repl.id of the
+        // target database to the event id of the last event considered by the dump. Next
+        // incremental cycle if starts from this id, the events considered for this dump, won't
+        // be considered again.
+
+        // The name of the database to be loaded into is either specified directly or is
+        // available from the dump metadata.
+        String dbName = work.dbNameToLoadIn;
+        if (dbName == null || StringUtils.isNotBlank(dbName)) {
+          if (work.currentReplScope != null) {
```

Comment:
Add a comment about in which scenario we hit this case.

Reply:
Done. Please check and suggest improvement if necessary.